uniswap_edpr/src/logics/shared_algorithm_lens/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+use shared_algorithm_lens_accessors :: * ;
+
+# [derive (Clone , Debug , Default , candid :: CandidType , serde :: Deserialize , serde :: Serialize)]
+pub struct LensValue {
+  pub address: String,
+  pub current_price: f32,
+  pub range_top: f32,
+  pub range_bottom: f32,
+  pub edpr: f32
+}
+
+pub async fn calculate (targets : Vec < String >) -> LensValue {
+  let pool_fees_result = get_get_last_snapshot_value_in_pool_fees (targets . get (0usize) . unwrap () . clone ()) . await ;
+  let tc28x6_result = get_get_last_snapshot_value_in_tcumul_28x6hr (targets . get (1usize) . unwrap () . clone ()) . await ;
+  let v3pool_result = get_get_last_snapshot_value_in_v3pool (targets . get (2usize) . unwrap () . clone ()) . await ;
+  let eth_usdc_price_result = get_get_last_snapshot_value_in_eth_usdc_price (targets . get (3usize) . unwrap () . clone ()) . await ;
+  
+  let fees_24h_usd = pool_fees_result.clone().unwrap().result[0].fees_24h_usd;
+  let t0_decimals = pool_fees_result.clone().unwrap().result[0].pool_summary_level_1.t0_decimals;
+  let t1_decimals =  pool_fees_result.clone().unwrap().result[0].pool_summary_level_1.t1_decimals;
```

Comment:
The result of the obtained data is unwrapped and cloned for each necessary field. This is a redundant process.

```rust
let pool_fees_result_0 = pool_fees_result.clone().unwrap().result[0]
let fees_24h_usd = pool_fees_result_0.fees_24h_usd;
let t0_decimals = pool_fees_result_0.t0_decimals;
let t1_decimals = pool_fees_result_0.t1_decimals;
```

Other results are the same.
Originally, destruct would be cleaner, but since structures are not explicit, the above should suffice.

Reply:
This was tricky, and took quite a while to get working correctly. I need to get better at Rust (that's the next part of my plan!) but I can see how your solution is much more efficient and eloquent.

Reply:
This is a code improvement, but it is something that should be addressed if it can be addressed.
Please indicate whether or not you are willing to respond.

Reply:
Not willing. It took quite a while to get these lines to compile. I agree with your reasoning but I tend to find that once I start changing these things it usually sets off a trail of error/help messages that takes at least a couple of hours to get through.

Reply:
I've made this change, and done the following for v3pool further on:
```
  let v3pool_result_res = v3pool_result.clone().unwrap().result;
  let current_tick_liquidity_str = v3pool_result_res.liquidity;
  /// and so on
```

I haven't yet pushed it or tried to build it. From what I remember from earlier in the development process, this might result in error messages saying that I have already used up the `pool_fees_result_0` variable once it is accessed for the second time (or words to that effect). If that does happen, I can't think of a more efficient way to do this. I think that may have been why I originally put in all the clone().unwrap()s, but I'll try it and see how it goes.